web/app/tournament/tourny.py

```diff
@@ -169,3 +172,93 @@ def select_player(tournament_id, tournament_player_id):
         # flash("Could not delete player: {{ tp.name }}")
         return redirect(url_for('.players_index',
             tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/matches/', methods=['GET', 'POST'])
+def matches_index(tournament_id):
+    if request.method == "GET":
+        matches = Match.query.filter_by(tournament_id=tournament_id)
+        return render_template('tournament_match_index.html',
+        tournament_id=tournament_id,
+        matches=matches)
+
+    if request.method == 'POST':
```

Comment:
From our conversation on hangouts: It's totally reasonable if you want to default to the tournament parameters. However, I would still hand-off the pairing to a method that accepts

- A list of players
- The method of pairing
- Pairing options.

That will make it much easier to test -- you needn't always make mock POST requests.